contracts/koh/contracts/KingAutomaton.sol

```diff
@@ -212,18 +218,23 @@ contract KingAutomaton {
     return id;
   }
 
-  function createProposal(address payable contributor, string calldata title, string calldata documents_link,
-      bytes calldata documents_hash) external returns (uint256) {
+  function createProposal(address payable contributor, string calldata title,
+      string calldata documents_link, bytes calldata documents_hash,
+      uint256 budget_period_len, uint256 num_periods, uint256 budget_per_period) external returns (uint256) {
+    // require(num_periods * budget_per_period <= SOME_LIMIT);
```

Comment:
SOME_LIMIT should really be a % of the available treasury balance. and that % should be set in the constructor as a parameter.